### PR TITLE
Adds soft_delete configuration option

### PR DESCRIPTION
**Pull request by abitdodgy**
@jnicklas I don't expect this to get merged, but I would like your feedback so I can improve the feature. I added a config option to `refile.rb` called `soft_delete`; it defaults to `false`. I check for this option before running line https://github.com/refile/refile/blob/master/lib/refile/attachment/active_record.rb#L44, when then `after_delete` callback executes. See issue #309 for reference.

Some feedback would be nice...

1. It feels wrong to add `soft_delete` to the config when it only works with ActiveRecord. It seems like `soft_delete` should work regardless of the adapter. The only other place I can think of implementing this is https://github.com/refile/refile/blob/master/lib/refile/attacher.rb#L134. Perhaps I'm wrong.
2. Should there be more tests? For example, that `soft_delete` defaults to `false` and that it can be set? Where would we add those? `refile_spec.rb`?
3. Is the one test in `spec/refile/attachment/active_record_spec.rb` sufficient for this feature?
4. Is `soft_delete` a good name?

### CODE REVIEW

**Comment by jnicklas** [comment]
1. I would rather have this as a config option to the `attachment` method, rather than a global config.
2. Yes, a few more tests would be good, I think `spec/refile/attachment/active_record_spec.rb` is the right place.
4. I've wanted this kind of option for a while (in fact I though it already existed and was surprised to find out it didn't), but I think it's more general than soft deletion. Someone might want to keep files around just in case, without implementing a soft deletion strategy. I think this is quite common. Some options could be:

``` ruby
attachment :foo, old_files: :keep # explicit but a bit wordy
attachment :foo, prune: true # difficult to understand, assumes the user knows what prune means
attachment :foo, destroy: true # destroy what exactly?
attachment :foo, dependent: :delete # Rails-alike, but maybe a bit confusing?
```

Not sure I really super like any of these.

**Comment by abitdodgy** [comment]
@jnicklas thanks for the feedback. I like `attachment :foo, dependent: :delete`, but I think it can be confusing.  IMO `old_files: :keep` is too verbose. How do you feel about:

```ruby
attachment :foo, keep_file: true    # seems explicit enough
attachment :foo, file: :destroy
# or, depending on which default behaviour we want... ?
attachment :foo, file: :destroy
```
Also, should `file` not be singular? We're dealing with one.

**Comment by jnicklas** [comment]
After thinking about this some more, I think that maybe I like `destroy: true|false` the best. What do you think?

**Comment by abitdodgy** [comment]
I think that sounds good! I'll start a new branch, and close this one once I make a new pull request.

**Comment by abitdodgy** [comment]
Closing in favour of #362
